keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,962 @@
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `pkg/api/pod`: `2023-01-19` - `72.8%`
+- `pkg/apis/core/validation`: `2023-01-19` - `82.9%`
+- `pkg/scheduler`: `2023-01-19` - `77.1%`
+- `k8s.io/kubernetes/pkg/scheduler/framework/plugins/defaultpreemption`: `2023-01-19` - `85.4%`
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- These cases will be added in the existed integration tests:
+  - Feature gate enable/disable tests
+  - During scheduling, `PreemptionPolicy` in `PodDisruptionBudget` works as expected
+  - Verify no significant performance degradation
+
+- `k8s.io/kubernetes/kubernetes/test/integration/scheduler/preemption/preemption_test.go`: https://storage.googleapis.com/k8s-triage/index.html?test=TestPreemption
+- `k8s.io/kubernetes/test/integration/scheduler_perf/scheduler_perf_test.go`: https://storage.googleapis.com/k8s-triage/index.html?test=BenchmarkPerfScheduling
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- These cases will be added in the existed e2e tests:
+  - Feature gate enable/disable tests
```

Comment:
We don't generally have this in E2E tests, as it's not easy to change configurations of the cluster.

But also we might be ok with just integration tests, as there are no more dependencies that scheduler and apiserver

Reply:
Ok. Updated.